### PR TITLE
Update CMake to 3.17.1

### PR DESCRIPTION
**Pull request by uilianries**
Changelog: (Feature | Fix | Bugfix): Describe here your pull request

- [ ] Refer to the issue that supports this Pull Request.
- [ ] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [ ] I've read the [Contributing guide](https://github.com/conan-io/conan-docker-tools/blob/master/.github/CONTRIBUTING.md).
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008) style guides for Python code.
- [ ] I've followed the [Best Practices](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/) guides for Dockerfile.

### CODE REVIEW

**Comment by Croydon** [comment]
So it only works, when nothing else builds? 😆 🙈

**Comment by uilianries** [comment]
build of schrodinger, if you don't look, it works.

File: conan_test_azure/Dockerfile [hunk 1]
```diff
@@ -42,19 +42,22 @@ RUN dpkg --add-architecture i386 \
        python3.8-dev \
        golang \
        pkg-config \
-       && rm -rf /var/lib/apt/lists/* \
-       && wget -q --no-check-certificate https://cmake.org/files/v${CMAKE_VERSION_MAJOR_MINOR}/cmake-${CMAKE_VERSION_FULL}-Linux-x86_64.tar.gz \
-       && tar -xzf cmake-${CMAKE_VERSION_FULL}-Linux-x86_64.tar.gz \
+       && rm -rf /var/lib/apt/lists/*
+
+RUN wget -q --no-check-certificate -O /tmp/cmake-${CMAKE_VERSION_FULL}-Linux-x86_64.tar.gz https://cmake.org/files/v${CMAKE_VERSION_MAJOR_MINOR}/cmake-${CMAKE_VERSION_FULL}-Linux-x86_64.tar.gz \
```
Comment (Croydon) [inline conan_test_azure/Dockerfile:47]: These env vars are not defined for this image

Comment (uilianries) [reply conan_test_azure/Dockerfile:47]: ouch!

Comment (Croydon) [reply conan_test_azure/Dockerfile:47]: Sorry for having bothering you for such a stupid bug 🙈

Comment (uilianries) [reply conan_test_azure/Dockerfile:47]: What?! You found the bug first hahah it's not your fault, we should found it at first commit (my fault)

Comment (Croydon) [reply conan_test_azure/Dockerfile:47]: Yeah, but I introduced this bug in my PR 

Probably with one of my mass search-and-replaces

**Pull request closed without merging**